flink/v1.12/flink/src/main/java/com/netease/arctic/flink/write/hidden/LogMsgFactory.java

```diff
@@ -41,13 +43,13 @@ Producer<T> createProducer(
   Consumer<T> createConsumer();
 
   interface Producer<T> {
-    void open() throws Exception;
+    void open(StreamingRuntimeContext context) throws Exception;
```

Comment:
I found that the Pulsar Producer only uses the context variable. 
How about adding a new open() method in the Producer interface?